src/CGIHandler.cpp

```diff
@@ -0,0 +1,352 @@
+		argv.push_back(&(*iter).at(0));
+	argv.push_back(NULL);
+}
+
+// HELPER FUNCTIONS
+
+/**
+ * @brief Extracts the path information from a given path based on the CGI extension.
+ *
+ * This function searches for the CGI extension within the provided path. If the extension
+ * is found, it then looks for the first occurrence of a '/' character after the extension.
+ * The substring starting from this '/' character is considered the path information and is returned.
+ * If the CGI extension or the '/' character is not found, an empty string is returned.
+ *
+ * @param path The full path from which to extract the path information.
+ * @return A string containing the path information if found, otherwise an empty string.
+ */
+std::string CGIHandler::extractPathInfo(const std::string& path)
+{
+	size_t extensionStart = path.find(m_cgiExt);
+	if (extensionStart == std::string::npos)
+		return "";
+	std::string tmp = path.substr(extensionStart);
+	size_t pathInfoStart = tmp.find('/');
+	if (pathInfoStart == std::string::npos)
+		return "";
+	return tmp.substr(pathInfoStart);
+}
+
+/**
+ * @brief Extracts the script path from the given path based on the CGI extension.
+ *
+ * This function searches for the CGI extension within the provided path and
+ * returns the substring from the beginning of the path up to and including the
+ * CGI extension. If the CGI extension is not found, an empty string is returned.
+ *
+ * @param path The full path from which to extract the script path.
+ * @return std::string The extracted script path or an empty string if the CGI extension is not found.
+ */
+std::string CGIHandler::extractScriptPath(const std::string& path)
+{
+	size_t extensionStart = path.find(m_cgiExt);
+	if (extensionStart == std::string::npos)
+		return "";
+	return path.substr(0, extensionStart + m_cgiExt.size());
+}
+
+/**
+ * @brief Registers signal handlers for various signals.
+ *
+ * This function sets the default signal handlers for SIGINT, SIGTERM, and SIGQUIT,
+ * and ignores the SIGHUP signal. If any of the signal registrations fail, an error
+ * message is logged and the function returns false.
+ *
+ * @return true if all signal handlers are successfully registered, false otherwise.
+ */
+bool registerChildSignals()
+{
+	// NOLINTNEXTLINE(cppcoreguidelines-pro-type-cstyle-cast, performance-no-int-to-ptr)
+	if (std::signal(SIGINT, SIG_DFL) == SIG_ERR) {
+		LOG_ERROR << "failed to register " << signalNumToName(SIGINT) << ": " << std::strerror(errno);
+		return (false);
+	}
+
+	// NOLINTNEXTLINE(cppcoreguidelines-pro-type-cstyle-cast, performance-no-int-to-ptr)
+	if (std::signal(SIGTERM, SIG_DFL) == SIG_ERR) {
+		LOG_ERROR << "failed to register " << signalNumToName(SIGTERM) << ": " << std::strerror(errno);
+		return (false);
+	}
+
+	// NOLINTNEXTLINE(cppcoreguidelines-pro-type-cstyle-cast, performance-no-int-to-ptr)
+	if (std::signal(SIGHUP, SIG_IGN) == SIG_ERR) {
```

Comment:
Cause there is not much to be reloaded. That is what the SIGHUP would trigger, no?

Reply:
SIGHUP gets sent when the controlling terminal is closed.

Nginx uses the signal as a sign to reload config. It runs daemonized anyway so it doesn't really have a controlling terminal.

Ignoring a signal is not reset by execve() which means the child process will ignore SIGHUP even though our server may already be shutdown.

If we were to ignore a signal it should be SIGQUIT, as this performs the graceful shutdown. In this case we would want to keep the forked children running so we can finish serving the already established client connections.

Reply:
I see, thx for the explanation. 

If we leave out signals, this becomes irrelevant anyways. Or do you want to remove all signals except for SIGQUIT which is set to be ignored?

Reply:
I think it makes sense to ignore SIGQUIT, since then we could perform a graceful shutdown with alive children. 

However I don't know if I'm maybe overlooking something, or how we would test this. Maybe a child which just sleeps for 5 secs and then returns an answer. Then we could check if we receive the answer if we graceful shutdown (would do this in another PR).

Reply:
see above